user/tests/wiring/sleep20/sleep20.cpp

```diff
@@ -755,3 +755,34 @@ test(29_System_Sleep_With_Configuration_Object_Ultra_Low_Power_Mode_Wakeup_Execu
     // Make sure we reconnect back to the cloud
     assertTrue(waitFor(Particle.connected, CLOUD_CONNECT_TIMEOUT));
 }
+
+test(30_System_Sleep_With_Configuration_Object_Ultra_Low_Power_Mode_Bypass_Network_Off_Execution_Time) {
+    constexpr uint32_t SLEEP_DURATION_S = 3;
+    Serial.printf("    >> Device enters ultra-low power mode. Please reconnect serial after %ld s\r\n", SLEEP_DURATION_S);
+    Serial.println("    >> Press any key now");
+    while (Serial.available() <= 0);
+    while (Serial.available() > 0) {
+        (void)Serial.read();
+    }
+
+    Particle.disconnect();
+    waitUntil(Particle.disconnected);
+
+    time32_t enter = Time.now();
+    SystemSleepConfiguration config;
+    config.mode(SystemSleepMode::ULTRA_LOW_POWER)
```

Comment:
We should ideally be testing HIBERNATE sleep mode here instead, as that's the main thing we've fixed.

Reply:
Hibernate mode will reset the device on waking up and device then waits for user to give instructions to start the test. So the elapsed time is not accurate. If needed, we can add a separated test case to verify that the hibernate mode does accept the network flag. What's your thought?

Reply:
1. You can use a retained variable to keep the start time
2. You can sample Time.now() in `STARTUP()` block as early as possible on boot.

I'd still like to have some sort of a test for HIBERNATE specifically ideally.